core/services/ocr2/plugins/ocr2vrf/coordinator/coordinator.go

```diff
@@ -358,9 +419,40 @@ func (c *coordinator) filterUnfulfilledCallbacks(
 	fulfilledRequestIDs map[uint64]struct{},
 	confirmationDelays map[uint32]struct{},
 	currentHeight int64,
+	currentBatchGasLimit int64,
 ) (callbacks []ocr2vrftypes.AbstractCostedCallbackRequest) {
-	// TODO: check if subscription has enough funds (eventually)
+
+	/**
+	 * Callback batch ordering:
+	 * - Callbacks are first ordered by beacon output + confirmation delay (ascending), in other words
+	 *   the fulfillments at the oldest block are first in line.
+	 * - Within the same block, fulfillments are ordered by gasAllowance (ascending), i.e the callbacks with
+	 *   the lowest gasAllowance are first in line.
```

Comment:
this seems like a good starting point to see how fast we can fulfill callbacks. But eventually, I think we need to also consider the age of the requests to avoid delaying fulfillments with high gas limits